plugin/keyring_vault/vault_memory.h

```diff
@@ -0,0 +1,47 @@
+#ifndef MYSQL_VAULT_MEMORY
+#define MYSQL_VAULT_MEMORY
+
+#include <my_global.h>
+#include <string.h>
+#include "keyring_memory.h"
+
+namespace keyring
+{
+#if !defined(HAVE_STDCXX11) && !defined(WIN32)
```

Comment:
I have changed it to check for memset_s function. I am using C++11 only for memset_s, so tt is cleaner if I check for memset_s, instead of C++11.